Expose more hf.co API surface, and ability to download a snapshot of a repo to a local folder (#25)
huggingface/huggingface_hub: README.md

Check out the source code for all possible params (we'll create a real doc page in the future).

> ### `snapshot_download`
> 
> `snapshot_download()` takes the following parameters, downloads the remote repository at the specified revision, 
> stores it to disk (in a versioning-aware) way and returns its local file path.
> 
> Parameters:
> - a `repo_id` in the format `namespace/repository`
> - a `revision` on which the repository will be downloaded
> - a `cache_dir` which you can specify if you want to control where on disk the files are cached.

Author: ```suggestion
### Bonus: `snapshot_download`

`snapshot_download()` downloads all the files from the remote repository at the specified revision, 
stores it to disk (in a versioning-aware way) and returns its local file path.

Parameters:
- a `repo_id` in the format `namespace/repository`
- a `revision` on which the repository will be downloaded
- a `cache_dir` which you can specify if you want to control where on disk the files are cached.
```
